### PR TITLE
synthetic subAttr referencing nested fields

### PR DESCRIPTION
**Pull request by geoberle**
the `subAttr` field of a synthetic declaration is used as a foreign key to define the backreference relation to another schema.

this PR enables the `subAttr` field to declare this relation not only to top level fields of the referenced schema but also nested fields, using `.` (dot) as a separator for nesting levels.

Example

```
- name: StatusPage_v1
  fields:
  - name: components
    type: StatusPageComponent_v1
    isList: true
    synthetic:
      schema: /dependencies/status-page-component-1.yml
      subAttr: some.path.to.a.page
```

Will be used in https://github.com/app-sre/qontract-schemas/pull/189 to backreference between maintenances and the statuspage components that participate.

Signed-off-by: Gerd Oberlechner <goberlec@redhat.com>

### CODE REVIEW

File: test/synthetic/synthetic.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      `;
+    const resp = await chai.request(srv)
+                        .post('/graphql')
+                        .set('content-type', 'application/json')
+                        .send({ query });
+    resp.should.have.status(200);
+    resp.body.data.test[0].name.should.equal('Pixiedust');
+    return resp.body.data.test[0].recipes[0].name.should.equal('Guillaumes Delux');
+  });
+});
```
[the remark this answers is not in the document]
Comment (geoberle) [reply test/synthetic/synthetic.test.ts:43]: Oops I thought there would be any but I can't find them either. 

I will create tests for the old functionality.

[the remark this answers is not in the document]
Comment (geoberle) [reply test/synthetic/synthetic.test.ts:63]: PTAL [c7b62af](https://github.com/app-sre/qontract-server/pull/150/commits/c7b62af00f8aad9d596bac65cc32de2a2ad5111b)

**Comment by bhushanthakur93** [comment]
LGTM

**Comment by bhushanthakur93** [comment]
Let's also update the schema documentation in this PR after we merge it?

**Comment by geoberle** [comment]
> Let's also update the schema documentation in this PR after we merge it?

added docs

**Review by rporres — APPROVED** [review]

**Pull request merged**